lib/jwt/jwk/rsa.rb

```diff
@@ -27,7 +27,8 @@ def public_key
       end
 
       def export(options = {})
-        exported_hash = members.merge(kid: kid)
+        kid # Make sure a kid is generated
```

Comment:
Could we avoid calling the kid method just to populate the parameters somehow. Would it be bad to generate the kid on initialization of the class?

Reply:
I was thinking about this as well, but thought at the time that there may be applications which would rather not spend the time computing a kid.

Thinking about it again now, the only flow in which a kid wouldn't be generated anyway is when importing a JWK somehow (probably usually via the `jwks` JWT decoding parameter) and later on only accessing the key pair (i.e. never exporting it). But those cases will most likely always use JWKSs from some sort of public interface, which _should_ already have `kid`s. And if some application really was this limited on resources, they could always replace the `kid` generating algorithm by some dummy alg. (And they probably wouldn't use Ruby for time-critical applications.)

So yes, populating it upon initialization seems like a good idea after all :)